tests/base_tests.py

```diff
@@ -239,6 +263,118 @@ def test_unexpected_kwarg_error(
             )
 
 
+class WeightColumnFitTests(GenericFitTests):
+    def test_fit_returns_self_weighted(
+        self,
+        minimal_attribute_dict,
+        uninitialized_transformers,
+    ):
+        """Test fit returns self?."""
+        df = d.create_df_9()
+        args = minimal_attribute_dict[self.transformer_name].copy()
+        args["weights_column"] = "c"
+
+        transformer = uninitialized_transformers[self.transformer_name](**args)
+
+        x_fitted = transformer.fit(df)
+
+        assert (
+            x_fitted is transformer
+        ), f"Returned value from {self.transformer_name}.fit not as expected."
+
+    def test_fit_not_changing_data_weighted(
+        self,
+        minimal_attribute_dict,
+        uninitialized_transformers,
+    ):
+        """Test fit does not change X - when weights are used."""
+        df = d.create_df_9()
+
+        args = minimal_attribute_dict[self.transformer_name].copy()
+        args["weights_column"] = "c"
+
+        transformer = uninitialized_transformers[self.transformer_name](**args)
+
+        transformer.fit(df)
+        ta.equality.assert_equal_dispatch(
+            expected=d.create_df_9(),
+            actual=df,
+            msg=f"X changed during fit for {self.transformer_name}",
+        )
+
+    @pytest.mark.parametrize(
+        "bad_weight_value, expected_message",
+        [
+            (np.nan, "weight column must be non-null"),
+            (np.inf, "weight column must not contain infinite values."),
+            (-np.inf, "weight column must be positive"),
+            (-1, "weight column must be positive"),
+        ],
+    )
+    def test_bad_values_in_weights_error(
```

Comment:
Some repeats in here with the below test

Reply:
Yeah can see that null and positive weight value checks are already in test_bad_values_in_weights_error. So have removed these checks from test_weight_not_in_X_error in [93740c8](https://github.com/lvgig/tubular/pull/239/commits/93740c86962fd004928463a3fce6e0dbe2060d99) as it makes more sense to have these as part of test_bad_values_in_weights_error test. Also, in case you were referring to repeats in terms of having both (-np.inf, "weight column must be positive") and (-1, "weight column must be positive") checks in test_bad_values_in_weights_error, it's because these checks are for different purpose: negative and -inf values.
Let me know if all the above makes sense :)

Reply:
thanks this is all good :)